ldap/servers/slapd/control.c

```diff
@@ -358,6 +467,25 @@ get_ldapmessage_controls_ext(
         managedsait = slapi_control_present(ctrls,
                                             LDAP_CONTROL_MANAGEDSAIT, NULL, NULL);
         slapi_pblock_set(pb, SLAPI_MANAGEDSAIT, &managedsait);
+        if (slapi_control_present(ctrls,
+                                  LDAP_CONTROL_X_SESSION_TRACKING, &session_tracking_spec, NULL)) {
+
+            parse_rc = parse_sessiontracking_ctrl(session_tracking_spec, &session_tracking_id);
+            if (parse_rc != LDAP_SUCCESS) {
+                Operation *pb_op = NULL;
+                slapi_pblock_get(pb, SLAPI_OPERATION, &pb_op);
+
+                slapi_log_err(SLAPI_LOG_WARNING, "get_ldapmessage_controls_ext", "Warning: conn=%" PRIu64 " op=%d failed to parse SessionTracking control (%d)\n",
```

Comment:
IMHO the message should be a warning if criticallity flag is false and an error otherwise (in this case the operation should return LDAP_UNAVAILABLE_CRITICAL_EXTENSION